src/tls.rs

```diff
@@ -251,28 +259,30 @@ pub(crate) struct TlsStream<T> {
     io: AllowStd<T>,
     is_shutdown: bool,
     session: ServerSession,
+    is_tls: bool,
 }
 
 impl<T> TlsStream<T> {
```

Comment:
I could make a `TlsUpgradeStream` if these `if` checks are disturbing performance.

Reply:
With the newest changes I suppose this is moot.